variable_set.go

```diff
@@ -0,0 +1,255 @@
+	Global      bool   `jsonapi:"attr,global"`
+
+	// Relations
+	Workspaces []*Workspace           `jsonapi:"relation,workspaces,omitempty"`
+	Variables  []*VariableSetVariable `jsonapi:"relation,vars,omitempty"`
+}
+
+type VariableSetListOptions struct {
+	ListOptions
+}
+
+func (o VariableSetListOptions) valid() error {
+	return nil
+}
+
+// List all Variable Sets in the organization
+func (s *variableSets) List(ctx context.Context, organization string, options VariableSetListOptions) (*VariableSetList, error) {
+	if !validStringID(&organization) {
+		return nil, ErrInvalidOrg
+	}
+	if err := options.valid(); err != nil {
+		return nil, err
+	}
+
+	u := fmt.Sprintf("organizations/%s/varsets", url.QueryEscape(organization))
+	req, err := s.client.newRequest("GET", u, &options)
+	if err != nil {
+		return nil, err
+	}
+
+	vl := &VariableSetList{}
+	err = s.client.do(ctx, req, vl)
+	if err != nil {
+		return nil, err
+	}
+
+	return vl, nil
+}
+
+// VariableSetCreateOptions represents the options for creating a new variable set within in a organization.
+type VariableSetCreateOptions struct {
+	// Type is a public field utilized by JSON:API to
+	// set the resource type via the field tag.
+	// It is not a user-defined value and does not need to be set.
+	// https://jsonapi.org/format/#crud-creating
+	Type string `jsonapi:"primary,vars"`
+
+	// The name of the variable set.
+	// Affects variable precedence when there are conflicts between Variable Sets
+	// https://www.terraform.io/cloud-docs/api-docs/variable-sets#apply-variable-set-to-workspaces
+	Name *string `jsonapi:"attr,name"`
+
+	// A description to provide context for the variable set.
+	Description *string `jsonapi:"attr,description"`
+
+	// If true the variable set is considered in all runs in the organization.
+	Global *bool `jsonapi:"attr,global"`
+}
+
+func (o VariableSetCreateOptions) valid() error {
+	if !validString(o.Name) {
+		return ErrRequiredName
+	}
+	if o.Global == nil {
+		return errors.New("global flag is required")
+	}
+	return nil
+}
+
+// Create is used to create a new variable set.
+func (s *variableSets) Create(ctx context.Context, organization string, options VariableSetCreateOptions) (*VariableSet, error) {
+	if !validStringID(&organization) {
+		return nil, ErrInvalidOrg
+	}
+	if err := options.valid(); err != nil {
+		return nil, err
+	}
+
+	u := fmt.Sprintf("organizations/%s/varsets", url.QueryEscape(organization))
+	req, err := s.client.newRequest("POST", u, &options)
+	if err != nil {
+		return nil, err
+	}
+
+	vl := &VariableSet{}
+	err = s.client.do(ctx, req, vl)
+	if err != nil {
+		return nil, err
+	}
+
+	return vl, nil
+}
+
+// Read is used to inspect a given variable set based on ID
+func (s *variableSets) Read(ctx context.Context, variableSetID string) (*VariableSet, error) {
+	if !validStringID(&variableSetID) {
+		return nil, errors.New("invalid variable set ID")
+	}
+
+	u := fmt.Sprintf("varsets/%s?include=workspaces", url.QueryEscape(variableSetID))
```

Comment:
Usually we allow users to specify include params using ReadOptions and ListOptions, which populates the relation fields in the struct. In the 1.0 release we'll be [making these go typed strings](https://github.com/hashicorp/go-tfe/pull/294)